ReSharper.FSharp/src/FSharp/FSharp.Psi.Daemon/src/Stages/PatternTypeHintsStage.fs

```diff
@@ -51,12 +63,18 @@ type private MembersVisitor(settings) =
         | :? IConstructorDeclaration -> true
         | _ -> false
 
+    let isLocalBinding (node: ITreeNode) =
+        match node with
+        | :? ILocalBinding
+        | :? ILambdaExpr -> true
```

Comment:
This is a corner case, but in some other cases we do consider these patterns to be somewhat top-level:
```fsharp
module Module

let f1 = fun x -> x + 1
let f1 x = fun y -> x + y
```

Reply:
What do you think about doing the same logic for this case here in a separate PR?

Reply:
Sure, as you wish 🙂